use-package.el

```diff
@@ -463,13 +463,11 @@ manually updated package."
 
 (defun use-package-handler/:ensure (name keyword ensure rest state)
   (let ((body (use-package-process-keywords name rest state)))
-    ;; This happens at macro expansion time, not when the expanded code is
-    ;; compiled or evaluated.
-    (let ((package-name (or (and (eq ensure t) (use-package-as-symbol name)) ensure)))
-      (when package-name
-        (require 'package)
-        (use-package-ensure-elpa package-name)))
-    body))
+    `((let ((package-name (or (and (eq ,ensure t) (use-package-as-symbol ',name)) ,ensure)))
```

Comment:
> which means that if you type (use-package test123| and do little pause between each letter, then it will call use-package-as-symbol many times and will intern symbols t, te, tes, test and so on.

Well, only in the case where you type `(use-package "test123`, but still a good point.

> `name` is a symbol, not a string.

No, I was talking about `package-name` (but FYI, `name` is *usually* symbol, sometimes a string). As in
```elisp
(let ((package-name-var (make-symbol "package-name")))
  `(let ((,package-name-var (or (and (eq ,ensure t) ...))))))
```
But actually, looking again, the `,@body` is ouside of `package-name`'s scope, so maybe it's fine to leave it as a normal symbol.